src/xdg.c

```diff
@@ -349,15 +349,25 @@ static void
 xdg_toplevel_view_move_to_front(struct view *view)
 {
 	struct view *root = xdg_toplevel_view_get_root(view);
+	/* FIXME: this exact code is repeated in xwayland.c */
 	view_impl_move_to_front(root);
 	view_impl_move_sub_views(root, LAB_TO_FRONT);
+	/* make sure view is in front of other sub-views */
+	if (view != root) {
+		view_impl_move_to_front(view);
+	}
 }
 
 static void
 xdg_toplevel_view_move_to_back(struct view *view)
 {
 	struct view *root = xdg_toplevel_view_get_root(view);
+	/* FIXME: this exact code is repeated in xwayland.c */
 	view_impl_move_sub_views(root, LAB_TO_BACK);
+	/* make sure view is behind other sub-views */
+	if (view != root) {
+		view_impl_move_to_back(view);
+	}
```

Comment:
The `move_to_front` LGTM.

The `move_to_back` part I am not so sure: when using the `Lower` action to move something to the back it causes a loss of the focused subview that was previously in front of all other (potentially rejecting input) subviews. When then clicking on the then-on-top subview (same for `A-Tab` or some panel click), we will raise the whole stack to the front again but may not be able to interact with it at all due to the application rejecting all input for other subviews and the toplevel itself. Can be tested with a text editor for example (I've used `GDK_BACKEND=x11 pluma`), opening the open-file dialog and then about dialog (which block input for everything else) then pressing a keybind that is set to run the `Lower` action. Result: the about dialog is between the editor and the open-file window and nothing can be interacted with unless manually focusing the about dialog again (which now works thanks to this PR). Doesn't work that easy when its hidden behind the new subview though.

So I don't think we need the `move_to_back` adjustment here as we just move the whole stack as it is to the back.

Reply:
That makes sense to me, thanks for thinking it through. I'll omit the `move_to_back` part.